staff_features/accessions/step_definitions/accession_spawn_digital_object.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+When 'the {string} setting is enabled in the Repository Preferences' do |repository_setting_checkbox_label|
+  find('#user-menu-dropdown').click
+  click_on 'Repository Preferences (admin)'
+
+  check repository_setting_checkbox_label
+
+  click_on 'Save'
+end
+
+Then 'the Create Digital Object modal is displayed' do
+  expect(page).to have_css '#accession_instances__0__digital_object__ref__modal'
```

Comment:
Not sure if I'm understanding this correctly but it seems a little error prone to check the css for presence of something on the page.  If what this is doing is checking to see if the included css has the class for a specific element id, wouldn't it be better to check that the html had the element with that id?

Reply:
I'm not sure what you mean by "see if the included css has the class for a specific element id." I think this is the usual Capybara way to do what is necessary here. `#accession_instances__0__digital_object__ref__modal` is just a selector that matches an element (a `div` in this case, which could be part of the selector but is probably not necessary) having that value in its `id` attribute, which I believe is what you are describing when you say "check that the html had the element with that id." CSS classes would not be involved as far as I know.

Reply:
@balmas 

> wouldn't it be better to check that the html had the element with that id?

This is what the Capybara matcher `have_css`.
It has nothing to do with any CSS staying.
I find it a rather confusing and bad name from their side, but we are stuck with it.